text/0745-user-provided-cdk-init-templates.md

```diff
@@ -0,0 +1,1678 @@
+
+```
+import * as cdk from 'aws-cdk-lib';
+import { Template } from 'aws-cdk-lib/assertions';
+import * as %name.PascalCased% from '../lib/%name%-stack';
+
+test('SQS Queue and SNS Topic Created', () => {
+  const app = new cdk.App();
+  const stack = new %name.PascalCased%.%name.PascalCased%Stack(app, 'MyTestStack');
+  const template = Template.fromStack(stack);
+
+  template.hasResourceProperties('AWS::SQS::Queue', {
+    VisibilityTimeoutSeconds: 300
+  });
+
+  template.hasResourceProperties('AWS::SNS::Topic', {});
+});
+```
+
+###### **Example Implementation of /tsconfig.json for  TypeScript template**
+
+```
+{
+  "compilerOptions": {
+    "target": "ES2022",
+    "module": "NodeNext",
+    "moduleResolution": "NodeNext",
+    "lib": [
+      "es2022"
+    ],
+    "declaration": true,
+    "strict": true,
+    "noImplicitAny": true,
+    "strictNullChecks": true,
+    "noImplicitThis": true,
+    "alwaysStrict": true,
+    "noUnusedLocals": false,
+    "noUnusedParameters": false,
+    "noImplicitReturns": true,
+    "noFallthroughCasesInSwitch": false,
+    "inlineSourceMap": true,
+    "inlineSources": true,
+    "experimentalDecorators": true,
+    "strictPropertyInitialization": false,
+    "typeRoots": [
+      "./node_modules/@types"
+    ]
+  },
+  "exclude": [
+    "node_modules",
+    "cdk.out"
+  ]
+}
+```
+
+###### **Example Implementation of /.template.gitignore for  TypeScript template**
+
+```
+*.js
+!jest.config.js
+*.d.ts
+node_modules
+
+# CDK asset staging directory
+.cdk.staging
+cdk.out
+```
+
+###### **Example Implementation of /README.md for  TypeScript template**
+
+```
+# %name.PascalCased% CDK App
+
+This is a starter AWS CDK app in TypeScript.
+
+## Getting Started
+
+```bash
+npm install
+npx cdk synth
+npx cdk deploy
+```
+
+NOTE: See Appendix A for example app template implementations in other CDK Languages.
+
+
+#### Testing Your Custom Templates Locally
+
+After authoring your custom templates, it can be helpful to test that `cdk init` runs successfully before publishing them to a Git repository or NPM package. Projects can be initialized from the authored template stored locally on your filesystem:
+
+```
+# Use a local template directory path
+$ cdk init --from-path ./my-cdk-template --language=[csharp|fsharp|go|java|javascript|python|typescript]
+
+# If template contains only one language, don't need to specify language
+$ cdk init --from-path ./my-cdk-template
+```
+
+
+Ticking the box below indicates that the public API of this RFC has been signed-off by the API bar raiser (the `status/api-approved` label was applied to the RFC pull request):
+
+```
+[ ] Signed-off by API Bar Raiser @xxxxx
+```
+
+
+
+## Public FAQ
+
+### What are we launching today?
+
+A new feature in the AWS CDK CLI that enables users to use custom templates for project initialization from local files, Git repositories, or NPM packages.
+
+
+### Why should I use this feature?
```

Comment:
What about discovery of templates? this seems like a benefit for both users and authors, but isn't mentioned.